### ISSUE TITLE
Create storage-glusterfs resources failed if external_provisioner enabled

### ISSUE THREAD

**Issue by qwang1** [kubevirt/kubevirt-ansible#206]
Hi there, 

I tried to provision storage-glusterfs with external_provisioner enabled. I met this problem when kubevirt-ansible executed "Create storage-glusterfs Resources":

unable to recognize "/tmp/storage-glusterfs.yml": no matches for /, Kind=ClusterRole
error validating "/tmp/storage-glusterfs.yml": error validating data: [ValidationError(ClusterRoleBinding): missing required field "userNames" in com.github.openshift.api.authorization.v1.ClusterRoleBinding, ValidationError(ClusterRoleBinding): missing required field "groupNames" in com.github.openshift.api.authorization.v1.ClusterRoleBinding]; if you choose to ignore these errors, turn validation off with --validate=false

Here it the ansible log:
```
[root@dhcp-140-0 kubevirt-ansible]# ansible-playbook --private-key /home/qwang/libra-new.pem -i inventory -e action=provision -e glusterfs_namespace=glusterfs -e storage_role=storage-glusterfs -e external_provisioner=true playbooks/storage.yml -v
Using /home/qwang/kubevirt-ansible-0423/kubevirt-ansible/ansible.cfg as config file
 [WARNING]: Found variable using reserved name: action


PLAY [masters[0]] *************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************
Monday 23 April 2018  18:51:04 +0800 (0:00:00.046)       0:00:00.046 ********** 
ok: [host-8-246-83.host.centralci.eng.rdu2.redhat.com]

TASK [storage-glusterfs : include_tasks] **************************************************************************************
Monday 23 April 2018  18:51:34 +0800 (0:00:30.362)       0:00:30.409 ********** 
included: /home/qwang/kubevirt-ansible-0423/kubevirt-ansible/roles/storage-glusterfs/tasks/provision.yml for host-8-246-83.host.centralci.eng.rdu2.redhat.com

TASK [storage-glusterfs : Login As Super User] ********************************************************************************
Monday 23 April 2018  18:51:34 +0800 (0:00:00.048)       0:00:30.458 ********** 
skipping: [host-8-246-83.host.centralci.eng.rdu2.redhat.com] => {
    "changed": false, 
    "skip_reason": "Conditional result was False"
}

TASK [storage-glusterfs : set_fact] *******************************************************************************************
Monday 23 April 2018  18:51:34 +0800 (0:00:00.026)       0:00:30.484 ********** 
ok: [host-8-246-83.host.centralci.eng.rdu2.redhat.com] => {
    "ansible_facts": {
        "glusterfs_name_label": "storage-"
    }, 
    "changed": false
}

TASK [storage-glusterfs : Determine the heketi route] *************************************************************************
Monday 23 April 2018  18:51:34 +0800 (0:00:00.036)       0:00:30.521 ********** 
changed: [host-8-246-83.host.centralci.eng.rdu2.redhat.com] => {
    "changed": true, 
    "cmd": [
        "oc", 
        "get", 
        "route", 
        "-n", 
        "glusterfs", 
        "-l", 
        "heketi=storage-route", 
        "-o", 
        "jsonpath={.items[0].spec.host}"
    ], 
    "delta": "0:00:00.314001", 
    "end": "2018-04-23 06:52:02.869695", 
    "rc": 0, 
    "start": "2018-04-23 06:52:02.555694"
}

STDOUT:

heketi-storage-glusterfs.apps.0422-dup.qe.rhcloud.com


TASK [storage-glusterfs : set_fact] *******************************************************************************************
Monday 23 April 2018  18:52:03 +0800 (0:00:29.493)       0:01:00.014 ********** 
ok: [host-8-246-83.host.centralci.eng.rdu2.redhat.com] => {
    "ansible_facts": {
        "heketi_url": "heketi-storage-glusterfs.apps.0422-dup.qe.rhcloud.com"
    }, 
    "changed": false
}

TASK [storage-glusterfs : Get heketi admin key] *******************************************************************************
Monday 23 April 2018  18:52:04 +0800 (0:00:00.062)       0:01:00.077 ********** 
changed: [host-8-246-83.host.centralci.eng.rdu2.redhat.com] => {
    "changed": true, 
    "cmd": "kubectl get pod -n glusterfs -l heketi=storage-pod -o jsonpath='{.items[0].spec.containers[0].env[?(@.name==\"HEKETI_ADMIN_KEY\")].value}' | base64 -", 
    "delta": "0:00:00.222785", 
    "end": "2018-04-23 06:52:30.459907", 
    "rc": 0, 
    "start": "2018-04-23 06:52:30.237122"
}

STDOUT:

T2M2bzk0TFZmcXhmZHFWRkNYQWw3WCtTMFc3SlZwUmlEOFpwWU5WclVLST0=


TASK [storage-glusterfs : set_fact] *******************************************************************************************
Monday 23 April 2018  18:52:31 +0800 (0:00:27.511)       0:01:27.589 ********** 
ok: [host-8-246-83.host.centralci.eng.rdu2.redhat.com] => {
    "ansible_facts": {
        "heketi_admin_key": "T2M2bzk0TFZmcXhmZHFWRkNYQWw3WCtTMFc3SlZwUmlEOFpwWU5WclVLST0="
    }, 
    "changed": false
}

TASK [storage-glusterfs : Render storage-glusterfs deployment yaml] ***********************************************************
Monday 23 April 2018  18:52:31 +0800 (0:00:00.061)       0:01:27.650 ********** 
ok: [host-8-246-83.host.centralci.eng.rdu2.redhat.com] => {
    "changed": false, 
    "checksum": "06dfdd364d784541edf0dc8c36df894269c18eec", 
    "gid": 0, 
    "group": "root", 
    "mode": "0644", 
    "owner": "root", 
    "path": "/tmp/storage-glusterfs.yml", 
    "secontext": "unconfined_u:object_r:admin_home_t:s0", 
    "size": 2600, 
    "state": "file", 
    "uid": 0
}

TASK [storage-glusterfs : Create storage-glusterfs Resources] *****************************************************************
Monday 23 April 2018  18:53:19 +0800 (0:00:47.998)       0:02:15.649 ********** 
fatal: [host-8-246-83.host.centralci.eng.rdu2.redhat.com]: FAILED! => {
    "changed": true, 
    "cmd": [
        "kubectl", 
        "apply", 
        "-n", 
        "glusterfs", 
        "-f", 
        "/tmp/storage-glusterfs.yml"
    ], 
    "delta": "0:00:00.466144", 
    "end": "2018-04-23 06:53:46.839006", 
    "rc": 1, 
    "start": "2018-04-23 06:53:46.372862"
}

STDOUT:

secret "kubevirt-heketi-secret" created
storageclass "kubevirt" created
serviceaccount "glusterfile-provisioner" created


STDERR:

unable to recognize "/tmp/storage-glusterfs.yml": no matches for /, Kind=ClusterRole
error validating "/tmp/storage-glusterfs.yml": error validating data: [ValidationError(ClusterRoleBinding): missing required field "userNames" in com.github.openshift.api.authorization.v1.ClusterRoleBinding, ValidationError(ClusterRoleBinding): missing required field "groupNames" in com.github.openshift.api.authorization.v1.ClusterRoleBinding]; if you choose to ignore these errors, turn validation off with --validate=false


MSG:

non-zero return code

	to retry, use: --limit @/home/qwang/kubevirt-ansible-0423/kubevirt-ansible/playbooks/storage.retry

PLAY RECAP ********************************************************************************************************************
host-8-246-83.host.centralci.eng.rdu2.redhat.com : ok=8    changed=2    unreachable=0    failed=1   

Monday 23 April 2018  18:53:47 +0800 (0:00:28.272)       0:02:43.922 ********** 
=============================================================================== 
storage-glusterfs : Render storage-glusterfs deployment yaml ---------------------------------------------------------- 48.00s
Gathering Facts ------------------------------------------------------------------------------------------------------- 30.36s
storage-glusterfs : Determine the heketi route ------------------------------------------------------------------------ 29.49s
storage-glusterfs : Create storage-glusterfs Resources ---------------------------------------------------------------- 28.27s
storage-glusterfs : Get heketi admin key ------------------------------------------------------------------------------ 27.51s
storage-glusterfs : set_fact ------------------------------------------------------------------------------------------- 0.06s
storage-glusterfs : set_fact ------------------------------------------------------------------------------------------- 0.06s
storage-glusterfs : include_tasks -------------------------------------------------------------------------------------- 0.05s
storage-glusterfs : set_fact ------------------------------------------------------------------------------------------- 0.04s
storage-glusterfs : Login As Super User -------------------------------------------------------------------------------- 0.03s
```

**Comment by jarrpa** [comment]
Could you provide the `/tmp/storage-glusterfs.yml` file?

**Comment by qwang1** [comment]
@jarrpa Hi,
```
[root@host-172-16-120-127 ~]# cat /tmp/storage-glusterfs.yml
---
apiVersion: v1
kind: Secret
metadata:
  name: kubevirt-heketi-secret
data:
  key: QVZtcmV4UjhKMnZoVlpxM1ZpYUpFRWY4TFlUMG42Y0VtOW0xK2tOZUNsbz0=
type: gluster.org/glusterfile
---
kind: StorageClass
apiVersion: storage.k8s.io/v1beta1
metadata:
  name: kubevirt
provisioner: gluster.org/glusterfile
reclaimPolicy: Delete
parameters:
  resturl: "http://heketi-storage-glusterfs.apps.0423-xdp.qe.rhcloud.com"
  restuser: admin
  secretName: "kubevirt-heketi-secret"
  secretNamespace: "glusterfs"
  volumeoptions: "group virt"
---
kind: ClusterRole
apiVersion: v1
metadata:
  name: glusterfile-provisioner-runner
  labels:
    glusterfs: file-provisioner-runner-clusterrole
    glusterfile: provisioner-runner-clusterrole
rules:
  - apiGroups: [""]
    resources: ["persistentvolumes"]
    verbs: ["get", "list", "watch", "create", "delete"]
  - apiGroups: [""]
    resources: ["persistentvolumeclaims"]
    verbs: ["get", "list", "watch", "update"]
  - apiGroups: ["storage.k8s.io"]
    resources: ["storageclasses"]
    verbs: ["get", "list", "watch"]
  - apiGroups: [""]
    resources: ["events"]
    verbs: ["list", "watch", "create", "update", "patch"]
  - apiGroups: [""]
    resources: ["services"]
    verbs: ["get","create","delete"]
  - apiGroups: [""]
    resources: ["secrets"]
    verbs: ["get"]
  - apiGroups: [""]
    resources: ["routes"]
    verbs: ["get", "list"]
  - apiGroups: [""]
    resources: ["endpoints"]
    verbs: ["get", "create","delete"]
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: glusterfile-provisioner
  labels:
    glusterfs: file-provisioner-sa
    glusterfile: provisioner-sa
---
apiVersion: v1
kind: ClusterRoleBinding
metadata:
  name: glusterfile-provisioner
roleRef:
  name: glusterfile-provisioner-runner
subjects:
- kind: ServiceAccount
  name: glusterfile-provisioner
  namespace: "glusterfs"
---
kind: Deployment
apiVersion: v1
metadata:
  name: glusterfile-provisioner
  labels:
    glusterfs: file-provisioner-deployment
    glusterfile: provisioner-deployment
  annotations:
    description: Defines how to deploy the glusterfile provisioner pod.
spec:
  replicas: 1
  strategy:
    type: Recreate
  template:
    metadata:
      name: glusterfile-provisioner
      labels:
        glusterfs: file-provisioner-pod
        glusterfile: provisioner-pod
    spec:
      serviceAccountName: glusterfile-provisioner
      containers:
      - name: glusterfile-provisioner
        image: gluster/glusterfileclone:latest
        imagePullPolicy: IfNotPresent
        env:
        - name: PROVISIONER_NAME
          value: gluster.org/glusterfile
```

**Comment by jarrpa** [comment]
Potentially fixed by #207.
